### ISSUE TITLE
Serialize information as YAML?

### ISSUE THREAD

**Issue by dannguyen** [danluu/debugging-stories#39]
This may be a case of over-engineering...but maybe it's worth filing each report with a specific structure and then generating the README from it? That way it's easier to skim the entries for the reader, and for the submitter, it's easier to know what's worth noting about a bug report.

For instance, the entry for "The case of the 500-mile email" could be serialized in YAML as:

~~~yaml
title: The case of the 500-mile email
url: http://www.ibiblio.org/harris/500milemail.html
archive_url: http://web.archive.org/web/20021220194409/http://www.ibiblio.org/harris/500milemail.html
author: Trey Harris
date: 2002-11-24
symptom: |
   Email failed to deliver to servers located further than 500 miles geographically from the sender.
cause: |
   Sendmail's default timeout value for remote connection is 0, which was practically equivalent to 3 ms -- i.e. roughly the time it takes to travel ~550 miles 

contextthingies:
      - name: Sendmail
        version: 5
~~~

Not sure what to call `contextthingies`...obviously, it can't just be "software" ("platform"?) ...and it seems prudent to make it a list of objects, since bug reports often involve the interlocking of software.

Anyway, I know it's significantly more overhead to add structure, but I think it might be a decent tradeoff for the flexibility of serialization in this case. Also, there's not that many debugging reports in the repo so far, so it won't take much time to serialize existing submissions. And submissions are relatively infrequent, so it's unlikely to be a huge barrier...

**Comment by danluu** [comment]
Yeah, I think that if this is going to get maintained it's pretty reasonable to have some kind of structured format that's used to generate the top level README, can be used to generate a nice looking HTML site, is searchable, etc.

If you want to submit a PR I'd be happy to take one, otherwise I'll get to it when I have some spare time (hahahaha...).
